core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractSample.java

```diff
@@ -505,4 +490,20 @@ public String getBarcodeSizeInfo() {
     return LimsUtils.makeVolumeAndConcentrationLabel(getVolume(), null, null);
   }
 
+  @Override
+  public String getDeleteType() {
+    return "Sample";
+  }
+
+  @Override
+  public String getDeleteDescription() {
+    return getName()
+        + (getAlias() == null ? "" : " (" + getAlias() + ")");
+  }
+
+  @Override
+  public SecurityProfile getDeletionSecurityProfile() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Update this method so that its implementation is not identical to "getSecurityProfile" on line 386. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4144)